x/tokenfactory/client/cli/query.go

```diff
@@ -84,8 +84,7 @@ func GetCmdDenomAuthorityMetadata() *cobra.Command {
 			}
 
 			res, err := queryClient.DenomAuthorityMetadata(cmd.Context(), &types.QueryDenomAuthorityMetadataRequest{
-				Creator:  denom[1],
-				Subdenom: denom[2],
+				Subdenom: args[0],
```

Comment:
it seems to be the full denom now, not a subdenom. afaik you need to update the `QueryDenomAuthorityMetadataRequest` proto message: 1) rename `subdenom` to `denom` and 2) remove `creator`

Reply:
it's not gonna happen.
request with `denom` instead of `subdenom` + `creator` has no pros, but has a decent con: you are forced to escape slash characters ( eg. `osmosis/tokenfactory/v1beta1/denoms/factory%2Fneutron1wzpddf4vr3plpeslsl4w45jr98spak0dpyf0v6%2Ftest3` compared to `osmosis/tokenfactory/v1beta1/denoms/factory/neutron1wzpddf4vr3plpeslsl4w45jr98spak0dpyf0v6/test3` 

so it's intentional

Reply:
no, I mean, behavior changed whereas field/params/vars names haven't: we used to pass `denom[1]` and `denom[2]` to the `types.QueryDenomAuthorityMetadataRequest` separately as `creator` and `subdenom` fields, but now we simply pass there the full `args[0]`, but the field name of the `types.QueryDenomAuthorityMetadataRequest` we assign this value to is still `subdenom` which is incorrect. additionally, the `types.QueryDenomAuthorityMetadataRequest`'s `creator` field seems not to be useful anymore